pyanaconda/anaconda.py

```diff
@@ -337,3 +340,24 @@ def writeXdriver(self, root=None):
         f = open("%s/etc/X11/xorg.conf" %(root,), 'w')
         f.write('Section "Device"\n\tIdentifier "Videocard0"\n\tDriver "%s"\nEndSection\n' % self.xdriver)
         f.close()
+
+    def run_dbus_if_required(self):
+        if not launcher.is_dbus_session_running():
+            launcher.start_dbus_session()
+
+    def run_boss(self):
+        bus_proxy = DBus.get_dbus_proxy()
+        bus_proxy.StartServiceByName(DBUS_BOSS_INSTALLATION_NAME,
+                                     DBUS_FLAG_NONE,
+                                     callback=self._check_boss_started)
```

Comment:
Does it really work when you call the method asynchronously? Because there is no loop yet.

Reply:
This callback is really called. Tested.
However, yeah it is interesting when GLib as you said is not running there. Maybe it is started by this?

Reply:
Or it might be called synchronously if there is no loop.